File: checker.py
# input.txt → avtor bergan input
# author.txt → avtor bergan output
# code.txt → userning kodi
# user.txt → user bergan output

ac = 0xAC
wa = 0xAD
pe = 0xAE
tl = 0xAF

author_input = open("input.txt", "r")
author_output = open("author.txt", "r")
user_output = open("user.txt", "r")
user_code = open("code.txt", "r")


def close(x):
    author_input.close()
    author_output.close()
    user_output.close()
    user_code.close()
    exit(x)


def main():
    # a + b misol uchun checker
    try:
        ans = int(author_output.readline())
        user_ans = int(user_output.readline())

        # user javob sifatida 1 tadan ortiq son chiqargan taqdirda
        if user_output.readline():
            close(pe)

        if ans == user_ans:
            close(ac)
        else:
            close(wa)
    except:
        close(pe)


if __name__ == "__main__":
    main()

File: checker.cpp
// input.txt → avtor bergan input
// author.txt → avtor bergan output
// code.txt → userning kodi
// user.txt → user bergan output
#include <bits/stdc++.h>

#define ac 0xAC
#define wa 0xAD
#define pe 0xAE
#define tl 0xAF

using namespace std;

typedef long long ll;
typedef unsigned long long ull;

ifstream input("input.txt");
ifstream output("author.txt");
ifstream user_output("user.txt");
ifstream code("code.txt");

void ret(int x) {
    input.close();
    output.close();
    user_output.close();
    code.close();
    exit(x);
}

int main()
{
    // a + b misol uchun checker
    try {
        ll ans, user_ans;
        output >> ans;
        user_output >> user_ans;

        // user javob sifatida 1 tadan ortiq son chiqargan taqdirda
        string s;
        if (user_output >> s) {
            ret(pe);
        }

        if (ans == user_ans) {
            ret(ac);
        }
        else {
            ret(wa);
        }

    }
    catch (const std::exception& e) {
        ret(pe);
    }
}
